Cache/Requests.lean

```diff
@@ -139,9 +139,15 @@ def downloadFiles (hashMap : IO.HashMap) (forceDownload : Bool) (parallel : Bool
   else IO.println "No files to download"
 
 /-- Check if the project's `lean-toolchain` file matches mathlib's.
-Print and error and exit the process with error code 1 otherwise. -/
+Print and error and exit the process with error code 1 otherwise.
+
+Does nothing if the current project is mathlib.
+-/
 def checkForToolchainMismatch : IO.CacheM Unit := do
-  let mathlibToolchainFile := (← IO.mathlibDepPath) / "lean-toolchain"
+  let mathlibToolchainFile := (← read).mathlibDepPath / "lean-toolchain"
+  if mathlibToolchainFile.normalize == ("lean-toolchain" : FilePath).normalize then
+    -- we are in mathlib, nothing to check
+    return ()
```

Comment:
I think `normalize` doesn't actually do what you expect and does not handle `./.` correctly